src/ChaiBuffer.hpp

```diff
@@ -294,6 +295,14 @@ class ChaiBuffer
   T * data() const
   { return m_pointer; }
 
+  /**
+   * @brief @return Return a pointer to the beginning of the buffer in a particular memory space.
+   * @param space The target memory space.
+   */
+  LVARRAY_HOST_DEVICE inline constexpr
+  T * data( MemorySpace const space ) const
+  { return static_cast< T * >( m_pointer_record->m_pointers[ internal::toChaiExecutionSpace( space ) ] ); }
```

Comment:
Is this sort of error prone? by this I mean that it seems likely that we can get a pointer that is not current? It there an underlying check/move?

Reply:
The code that uses this in GEOSX calls a move before. Would you rather have a call to `move` embedded in here? This sort of makes synchronization unavoidable, and I'm not sure we always want it.

Reply:
I'm fine with this for the use case you describe above you don't want to move the data before getting the pointers our. Obviously you could really shoot yourself but this should only be used in a few places.

Could you
1. Add a check to error out if the array wasn't allocated in that space `m_pointer_record->m_pointers[ space ] == nullptr`?
2. Add the following the `MallocBuffer` and `StackBuffer`?
``` c++
T * data( MemorySpace const space ) const
{
  LVARRAY_ERROR_IF_NE( space, MemorySpace::CPU );
  return data();
}
```
3. Add a simple test to `testArrayView`.

Reply:
@corbett5 done